src/Glpi/Api/Rest/Client.php

```diff
@@ -82,7 +83,7 @@ public function initSessionByCredentials($user, $password) {
       if ($response->getStatusCode() != 200
          || !$this->sessionToken = json_decode($response->getBody()->getContents(), true)['session_token']) {
          $body = json_decode($response->getBody()->getContents());
-         throw new Exception(ErrorHandler::getMessage($body[0]));
+         throw new BadEndpointException(ErrorHandler::getMessage($body[0]));
```

Comment:
The endpoint of this method is trivial and always exists. 

If the response is !== 200 then the er ror is not due to a bad endpoint. It would be more related to a bad request (malformed, or invalid credentials). Then the BadEndpointException is not a good name here.